src/middleware/analytics.ts

```diff
@@ -0,0 +1,85 @@
+import axios from 'axios';
+import { GA_TRACKING_ID, GA_API_SECRET, GA_URL, DEBUG_GA_URL, isProduction as isProductionConfig } from '../config';
+
+const trackEvent = ({ clientIdentifier, category, action, keyword }) =>
+  new Promise((resolve, reject) => {
+    const params = {
+      measurement_id: GA_TRACKING_ID,
+      api_secret: GA_API_SECRET,
+    };
+
+    const isProduction = isProductionConfig && params.measurement_id && params.api_secret;
+
+    const data = JSON.stringify({
+      client_id: clientIdentifier,
+      events: [
+        {
+          name: 'search',
+          params: {
+            category,
+            action,
+            search_term: keyword,
+          },
+        },
+      ],
+    });
+
+    // Avoid sending GA events
+    if (isProduction) {
+      console.time('Sending production tracking data');
+      axios({
+        method: 'post',
+        url: `${GA_URL}?measurement_id=${params.measurement_id}&api_secret=${params.api_secret}`,
+        data,
+      })
+        .then((res) => {
+          console.log('Logging the data:', JSON.parse(data), JSON.parse(data).events[0].params);
+          console.log({ status: res.status, response: res.data });
+          console.timeEnd('Sending production tracking data');
+          resolve(true);
+        })
+        .catch((err) => {
+          console.log(typeof err?.toJSON === 'function' ? err.toJSON() : err);
+          console.timeEnd('Sending production tracking data');
+          reject(new Error(err));
+        });
+    } else {
+      console.time('Sending development tracking data');
+      axios({
+        method: 'post',
+        url: `${DEBUG_GA_URL}?measurement_id=${params.measurement_id}&api_secret=${params.api_secret}`,
+        data,
+      })
+        .then(() => {
+          console.timeEnd('Sending development tracking data');
+          resolve(true);
+        })
+        .catch((err) => {
+          if (isProduction) {
+            console.log(typeof err?.toJSON === 'function' ? err.toJSON() : err);
+            console.timeEnd('Sending development tracking data');
+            reject(new Error(err));
+          }
+        });
+    }
+  });
+
+export default async (req, _, next) => {
+  try {
+    const { method } = req;
+    const developerAPIKey = req.headers['X-API-Key'] || req.headers['x-api-key'];
+    const { keyword } = req.query;
+    const { pathname } = req._parsedUrl;
```

Comment:
.... added the `'no-underscore-dangle': 'off',` eslint rule because of this line

Reply:
i don't think we should change the eslint rules for one case

let's add `// ts-expect-error eslint(no-underscore-dangle) so that we can address this error in a follow ticket
```suggestion
    // ts-expect-error eslint(no-underscore-dangle)
    const { pathname } = req._parsedUrl;
```

I did a little investigating and when I add `req: Request` it looks like `_parsedUrl` doesn't live on the request object, so this is a clear sign that we actually want to stop using `_parsedUrl` and change it to a typed equivalent

but for this PR, let's just add the ts-expect-error and you can take on the follow up ticket to change this variable (along side add types function arguments)

Reply:
For some reason, `ts-expect-error eslint(no-underscore-dangle)` does not work.
The compiler still throws an error.

But this works if added on that line:
/* eslint no-underscore-dangle: 0 */

Should I go ahead and make the changes?

Reply:
Sorry, it's not a TS error rather it's a ESLint error

You want to use the following comment above the line of code that has the error:
`// eslint-disable-next-line no-underscore-dangle`